shell/browser/net/atom_url_loader_factory.cc

```diff
@@ -78,7 +78,7 @@ bool ResponseMustBeObject(ProtocolType type) {
 
 // Helper to convert value to Dictionary.
 mate::Dictionary ToDict(v8::Isolate* isolate, v8::Local<v8::Value> value) {
-  if (value->IsObject())
+  if (!value->IsFunction() && value->IsObject())
```

Comment:
i'm confused what error this fixes?

Reply:
It fixes the test in https://github.com/electron/electron/blob/master/spec-main/api-protocol-spec.ts#L158 . `v8::Function` inherits from `v8::Object` so `IsObject` will be true and the eventual code path is not what we wanted https://v8docs.nodesource.com/node-10.6/d5/d54/classv8_1_1_function.html. In pre-network service code path it was taken care by `v8_value_converter` https://github.com/electron/electron/blob/master/shell/common/native_mate_converters/v8_value_converter.cc#L359 coincidentally where function type was checked before object type.

Reply:
Maybe I should have documented this